### ISSUE TITLE
Two queries are executed when visiting category page url without query parameters

### ISSUE THREAD

**Issue by Inwerpsel** [algolia/algoliasearch-magento#831]
When instantsearch on category pages is enabled then two queries are executed.

Steps to reproduce:
- go to [https://magento.algolia.com/accessories/eyewear.html](https://magento.algolia.com/accessories/eyewear.html)
- open network pane, filter on XHR and refresh
- there are two almost identical requests for the search

This seems to be caused by [https://github.com/algolia/algoliasearch-magento/blob/master/js/algoliasearch/instantsearch.js#L461-L467](https://github.com/algolia/algoliasearch-magento/blob/master/js/algoliasearch/instantsearch.js#L461-L467): first search.start() is called then search.search() is called.

**Comment by JanPetr** [comment]
Thanks for the report @Inwerpsel, I investigate and come back to you.

**Comment by Inwerpsel** [comment]
Wouldn't it make more sense to use the url to know if any other category filter is applied instead of executing the search just to know this?

Probably it is possible to set the category parameter after the search object is instantiated with on https://github.com/algolia/algoliasearch-magento/blob/master/js/algoliasearch/instantsearch.js#L66 but before calling search,start().

**Comment by JanPetr** [comment]
It'd be the best, but it has some unexpected behavior as it won't add category to current refinements and causes confusion for end users.
The second query is there to correctly filter the category and show it in current refinements widget.

**Comment by Inwerpsel** [comment]
```javascript
        if (algoliaConfig.request.path.length > 0 && window.location.hash.indexOf('categories.level0') === -1) {
		    instantsearchOptions.searchParameters = {
                hierarchicalFacetsRefinements: {
                    'categories.level0': [algoliaConfig.request.path]
                }
            }
        }

```

There is probably a better way to check the url if any other category filters have been applied, but this applies the current category filter before the search is made.

Then, to make the category appear in the current refinements, you can just remove the lines that exclude this facet when on a category page:

```javascript
        $.each(algoliaConfig.facets, function (i, facet) {
            var name = facet.attribute;

            if (name === 'categories') {
                // if (algoliaConfig.isCategoryPage) {
                //    return;
                // }
                name = 'categories.level0';
            }

            if (name === 'price') {
                name = facet.attribute + algoliaConfig.priceKey
            }

            attributes.push({
                name: name,
                label: facet.label ? facet.label : facet.attribute
            });
        });```

**Comment by Inwerpsel** [comment]
It appears that those lines I mentioned I had commented out where removed in recent versions of the module and I had added them again myself in the code of the project I'm using the module for at work.

However it was still the only thing I needed to do to have the category appear in the current refinements without having to call the index twice.

**Comment by Inwerpsel** [comment]
In addition to executing two queries the current code has another issue: it doesn't work well with browser history.

To reproduce:
- go to a category page
- select another category filter
- hit back

What you would expect is see that the original category filter is being applied, what actually happens is that no filter is applied.

If the filter is applied before executing the first query then going back in history does give the correct category filter.

**Comment by JanPetr** [comment]
Thank you for your work on that @Inwerpsel! ❤️  I created a fix  - #852. Could you give it a try and confirm that it works for you? Thanks a lot!

**Comment by JanPetr** [comment]
Fix will be part of the next release
